sycl/doc/extensions/experimental/sycl_ext_oneapi_annotated_ptr.asciidoc

```diff
@@ -476,8 +476,8 @@ a|
 T* get() const noexcept;
 ----
 |
-Returns the underlying raw pointer. The raw pointer will not retain the
-annotations.
+Returns the underlying raw pointer. Implementations are free to propagate information from properties of
+an annotated_ptr to the raw pointer.
```

Comment:
This raises an interesting question.  Is there a guaranteed way to get a raw pointer that does *not* have the annotations?  This might be important for an annotation that somehow changes the semantic of the program (i.e. not a "hint" annotation).

Reply:
Right now there is no member function/operator that guarantees to get the underlying raw pointer with none annotation attached. A possible solution for this is to enable explict conversion from `annotated_ptr<T>` to `T*`.

Also I'm wondering if it's really needed. If user explicitly creates an `annotated_ptr` instance with a non-hint annotation, is it legal to get the underlying pointer without that annotation?  Since not having this annotation changes the semantic, I would expect the compiler to issue a warning or even error if it detects the risk of losing the information.